Bug systemu konfiguracji. Niepoprawne pobieranie Stringa.

Powiązane: https://github.com/FunnyGuilds/FunnyGuilds/issues/1602

Startowa konfiguracja:
> guild-tnt-protection-start-time: '22:00'
> guild-tnt-protection-end-time: 06:00

Po wpisaniu godziny powyżej 09:59
np. 10:00 bez ' system konfiguracji podaje złego Stringa do parsowania.

Błąd dla przykładu z 10:00
```
[17:19:05 ERROR]: [FunnyGuilds]
[17:19:05 ERROR]: [FunnyGuilds] Could not load plugin configuration
[17:19:05 ERROR]: [FunnyGuilds]
[17:19:05 ERROR]: [FunnyGuilds] net.dzikoysk.funnyguilds.libs.eu.okaeri.configs.exception.OkaeriException: failed to initialize net.dzikoysk.funnyguilds.data.configs.PluginConfiguration [class net.dzikoysk.funnyguilds.libs.eu.okaeri.configs.validator.okaeri.OkaeriValidator]
[17:19:05 ERROR]: [FunnyGuilds]         at net.dzikoysk.funnyguilds.libs.eu.okaeri.configs.ConfigManager.create(ConfigManager.java:53)
[17:19:05 ERROR]: [FunnyGuilds]         at net.dzikoysk.funnyguilds.FunnyGuilds.onLoad(FunnyGuilds.java:104)
[17:19:05 ERROR]: [FunnyGuilds]         at org.bukkit.craftbukkit.v1_16_R3.CraftServer.loadPlugins(CraftServer.java:393)
[17:19:05 ERROR]: [FunnyGuilds]         at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:206)
[17:19:05 ERROR]: [FunnyGuilds]         at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:936)
[17:19:05 ERROR]: [FunnyGuilds]         at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:174)
[17:19:05 ERROR]: [FunnyGuilds]         at java.lang.Thread.run(Thread.java:748)
[17:19:05 ERROR]: [FunnyGuilds] Caused by: java.time.format.DateTimeParseException: Text '600' could not be parsed at index 2
[17:19:05 ERROR]: [FunnyGuilds]         at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949)
[17:19:05 ERROR]: [FunnyGuilds]         at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
[17:19:05 ERROR]: [FunnyGuilds]         at java.time.LocalTime.parse(LocalTime.java:441)
[17:19:05 ERROR]: [FunnyGuilds]         at net.dzikoysk.funnyguilds.data.configs.PluginConfiguration.load(PluginConfiguration.java:1359)
[17:19:05 ERROR]: [FunnyGuilds]         at net.dzikoysk.funnyguilds.libs.eu.okaeri.configs.OkaeriConfig.load(OkaeriConfig.java:204)
[17:19:05 ERROR]: [FunnyGuilds]         at net.dzikoysk.funnyguilds.FunnyGuilds.lambda$onLoad$2(FunnyGuilds.java:108)
[17:19:05 ERROR]: [FunnyGuilds]         at net.dzikoysk.funnyguilds.libs.eu.okaeri.configs.ConfigManager.create(ConfigManager.java:50)
[17:19:05 ERROR]: [FunnyGuilds]         ... 6 more
[17:19:05 ERROR]: [FunnyGuilds]

```

--- a/plugin/src/main/java/net/dzikoysk/funnyguilds/config/FunnyTime.java
+++ b/plugin/src/main/java/net/dzikoysk/funnyguilds/config/FunnyTime.java
@@ -5,7 +5,7 @@ import java.time.format.DateTimeFormatter;
 
 public class FunnyTime {
 
-    public static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");
+    public static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("H:mm");
 
     private final LocalTime time;
     private final String formattedTime;

--- a/plugin/src/main/java/net/dzikoysk/funnyguilds/config/transformer/FunnyTimeTransformer.java
+++ b/plugin/src/main/java/net/dzikoysk/funnyguilds/config/transformer/FunnyTimeTransformer.java
@@ -5,6 +5,7 @@ import eu.okaeri.configs.serdes.BidirectionalTransformer;
 import eu.okaeri.configs.serdes.SerdesContext;
 import java.time.LocalTime;
 import net.dzikoysk.funnyguilds.config.FunnyTime;
+import org.jetbrains.annotations.NotNull;
 
 public class FunnyTimeTransformer extends BidirectionalTransformer<String, FunnyTime> {
 
@@ -14,12 +15,44 @@ public class FunnyTimeTransformer extends BidirectionalTransformer<String, Funny
     }
 
     @Override
-    public FunnyTime leftToRight(String data, SerdesContext serdesContext) {
-        return new FunnyTime(LocalTime.parse(data, FunnyTime.TIME_FORMATTER));
+    public FunnyTime leftToRight(@NotNull String data, @NotNull SerdesContext serdesContext) {
+
+        // standard parse from string
+        if (data.contains(":")) {
+            return new FunnyTime(LocalTime.parse(data, FunnyTime.TIME_FORMATTER));
+        }
+
+        // something went wrong, probably yaml spec, whoopsie... https://noyaml.com/
+        int value;
+        try {
+            value = Integer.parseInt(data);
+        }
+        catch (NumberFormatException exception) {
+            throw new IllegalArgumentException("Cannot resolve time from " + data, exception);
+        }
+
+        // the value is above 24 hours in both cases
+        if (value > 86400) {
+            throw new IllegalArgumentException("Cannot resolve time from " + value);
+        }
+
+        // value in minutes is above 24 hours
+        // restore as seconds from midnight
+        // 00:00 - 00:24 may have bad time
+        if (value > 1440) {
+            int hours = value / 3600;
+            int minutes = (value % 3600) / 60;
+            return new FunnyTime(LocalTime.of(hours, minutes));
+        }
+
+        // restore as minutes from midnight
+        int hours = value / 60;
+        int minutes = value % 60;
+        return new FunnyTime(LocalTime.of(hours, minutes));
     }
 
     @Override
-    public String rightToLeft(FunnyTime data, SerdesContext serdesContext) {
+    public String rightToLeft(@NotNull FunnyTime data, @NotNull SerdesContext serdesContext) {
         return data.getFormattedTime();
     }
 

